packages/trader/src/App/Components/Elements/Errors/error-component.jsx

```diff
@@ -1,17 +1,33 @@
 import PropTypes from 'prop-types';
 import React from 'react';
-import { PageError } from '@deriv/components';
+import { PageError, Dialog } from '@deriv/components';
 import { routes } from '@deriv/shared';
-import { Localize } from '@deriv/translations';
+import { Localize, localize } from '@deriv/translations';
 
-const ErrorComponent = ({ header, message, redirect_label, redirectOnClick, should_show_refresh = true }) => {
+const ErrorComponent = ({
+    header,
+    message,
+    is_dialog,
+    redirect_label,
+    redirectOnClick,
+    should_show_refresh = true,
+}) => {
     const refresh_message = should_show_refresh ? (
         <Localize i18n_default_text='Please refresh this page to continue.' />
     ) : (
         ''
     );
 
-    return (
+    return is_dialog ? (
+        <Dialog
+            title={localize('There was an error')}
+            is_visible
+            confirm_button_text={localize('Ok')}
+            onConfirm={redirectOnClick || (() => location.reload())}
+        >
+            {message || <Localize key={0} i18n_default_text='Sorry, an error occured while processing your request.' />}
```

Comment:
Why not just use the `localize` function here instead?

Reply:
@zaki-binary the message comes from API, which already has a translation from the API, am I wrong?

Reply:
@zaki-binary just updated the code